src/plugins/auto/src/cumulative_schedule.cpp

```diff
@@ -10,6 +10,21 @@
 // ------------------------------CumuSchedule----------------------------
 namespace ov {
 namespace auto_plugin {
+std::string CumuSchedule::schedule_to_next_device(const std::vector<DeviceInformation>& devices,
+                                                  std::size_t current_device_index) {
+    std::lock_guard<std::mutex> lock(m_context->m_mutex);
+    m_n_ctput_schedule_nextdevice = m_n_ctput_schedule_nextdevice >= devices.size() ? 0 : m_n_ctput_schedule_nextdevice;
+    auto selected_device_name = devices[m_n_ctput_schedule_nextdevice].device_name;
+    auto schedule_policy = m_context->m_schedule_policy;
+    if (schedule_policy == ov::intel_auto::SchedulePolicy::ROUND_ROBIN) {
+        m_n_ctput_schedule_nextdevice++;
+    } else if (schedule_policy == ov::intel_auto::SchedulePolicy::DEVICE_PRIORITY) {
+        m_n_ctput_schedule_nextdevice = current_device_index;
+        selected_device_name = devices[m_n_ctput_schedule_nextdevice].device_name;
+    }
+    return selected_device_name;
+}
```

Comment:
some questions:
1) this property should support change on-the-fly? if so, please add related property setting/updating
2) if 1) is not true, I think for DEVICE_PRIORITY policy, we don't need to maintain and update m_n_ctput_schedule_nextdevice
also, the lock here, let's consider together with the runtime fallback case

Reply:
1. this property will not be changed on-the-fly
2. Updated.
3. Considering the runtime fallback case, I will keep the lock of  `m_context->m_fallback_mutex ` before schedule function is done

Reply:
Updated. No need to keep the lock of `m_context->m_fallback_mutex` during infer request schedule since no resources associated with the infer request queue are cleaned up during the runtime fallback process.